### PR TITLE
Call assignment stats in header

### LINKED ISSUE #836: Starting, ending and scheduling call assignments
## Description
A user should be able to set the start and end date of a call assignment.

## Relevant Job Stories
- When it’s time to start calling on an assignment, I want to be able to open it up for calling, so that it’s only available once it’s ready
- When we finish calling on an assignment, I want to close it so that all the callers lose access to it without having to remove them one by one
- When I know beforehand what days we’re phone banking, I want to e able to set the start and end date, so that it automatically opens and closes on the right days
- When I’m planning a call assignment to continue indefinitely, I want to be able to configure it without a close date, so that it doesn’t close until I do so manually

## Prerequisites
Requires #833 and #834 

## Requirements
- [ ] The call assignment header should include a date range picker (#833) to pick the start date and optional end date of a call assignment
- [ ] When not open/active, the header should include a button called "Start assignment"
- [ ] When open/active, the header should include a button called "End assignment"

### Button actions
The actions taken by the "Start assignment" and "End assignment" buttons depend on the current state of the assignment (the start/end dates specifically). See the following matrix ([spreadsheet](https://docs.google.com/spreadsheets/d/1_WnqMNYetrQImA0fTNPUSLiKd1HEkBZ_2rDi97lQp-c/edit#gid=0)) for details:

image

## Possible implementations
Use new ZUI component documented in #833.

## Design specifications
image

## Open questions
None

### LINKED ISSUE #837: Call assignment basic stats in header
## Description
A user should be able to assess the size of a call assignment by glancing over it's header.

## Relevant Job Stories
None

## Prerequisites
Requires #834 

## Requirements
- [ ] Number of callers should be displayed in header
- [ ] Number of targets should be displayed in header
  - [ ] If targeting has not been configured, hide widget

## Possible implementations
None

## Design specifications
image

## Open questions
* There are no job stories for this, so maybe it's not relevant? Low priority!

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR adds basic stats to the call assignment header.

## Screenshots
image

## Changes
* Adds new `ZUIFuture` component to conditionally render future (async loading) content
* Fixes issues with how callers were not reported as loading while loading (due to filtering logic)
* Displays target count and caller count in header

## Notes to reviewer
Please only review this after #855 has been merged.

## Related issues
Resolves #837

### CODE REVIEW

File: src/zui/ZUIFuture/index.tsx
Comment (ziggabyte) [inline src/zui/ZUIFuture/index.tsx:49]: I like this and look forward to using and exploring the potential this Future component has!

**Review by ziggabyte — APPROVED** [review]
When playing with it what I notice is that the skeleton seems to take up more height than the stats do when loaded, so the header does a little twitch to adjust to the smaller height. If this can be easily managed it would satisfy my eyes a lot, but if it's too complex it's good enough for now if it's about fx different screen sizes of our computers.

**Comment by richardolsson** [comment]
After talking to @ziggabyte we decided to merge this and revisit the skeleton once the component has seen more use cases.

**Pull request merged**
